recipes/datoviz/meta.yaml

```diff
@@ -0,0 +1,80 @@
+# Note: there are many handy hints in comments in this example -- remove them when you've finalized your recipe
+
+# Jinja variables help maintain the recipe as you'll update the version only here.
+# Using the name variable with the URL in line 14 is convenient
+# when copying and pasting from another recipe, but not really needed.
+{% set name = "datoviz" %}
+{% set version = "0.1.0" %}
+
+package:
+  name: {{ name|lower }}
+  version: {{ version }}
+
+source:
+  url: https://pypi.io/packages/source/{{ name[0] }}/{{ name }}/{{ name }}-{{ version }}.tar.gz
```

Comment:
@rossant -- how would you like to handle this? One option for now would be to pull from the alpha release on github.

Reply:
I don't know what the conda-forge policy is regarding alpha versions. Either pull from the alpha release on github, or wait until I upload to PyPI. For the latter to happen I'd just need some testing of a few users on various operating systems.

That being said, if we use this with conda-forge, it means we need another CI system in place for building the various wheels in the first place?

Reply:
I've been progressively adding the dependencies to the recipe. Do you all have a recommendation for how to install Vulkan? I guess we might need to package that into another recipe?

For context: https://dev.azure.com/conda-forge/feedstock-builds/_build/results?buildId=338645&view=logs&j=6f142865-96c3-535c-b7ea-873d86b887bd&t=22b0682d-ab9e-55d7-9c79-49f3c3ba4823&l=704

Reply:
Did you have a look at the Dockerfile_wheel? It shows how to install Vulkan on Linux

Reply:
Yeah, I did take a look. I guess we're going to need to create a separate package for `vulkan` according to one of their checklist items:

> Package does not vendor other packages. (If a package uses the source of another package, they should be separate packages or the licenses of all packages need to be packaged).

Reply:
Ah, this is going to be tricky. I don't think we can package vulkan. Datoviz is built with the Vulkan SDK. It is compiled with the Vulkan C headers, and linked with the Vulkan library provided by the SDK. When distributing Datoviz, I don't think the package should bundle Vulkan. Rather, the runtime should link to the user's Vulkan library that should be provided by the GPU drivers. I think this is how it ought to work on Windows and Linux.

There is a complication on macOS. There is no Vulkan on macOS, instead there is the proprietary Apple Metal API, and another project called MoltenVK which is now part of the [LunarG Vulkan SDK](https://vulkan.lunarg.com/sdk/home) for macOS. MoltenVK translates Vulkan calls into Metal, and this is how Datoviz can work on macOS.

Unfortunately, I haven't yet managed to build a pip wheel properly on macOS. One has to read the Vulkan SDK documentation in detail to understand how libvulkan should be bundled in the wheel. I tried bundling it, but then glfw doesn't manage to create a surface. If I don't bundle it, then it logically doesn't find libvulkan at runtime, which is normal since Vulkan doesn't exist on macOS unless you install the SDK. Requiring macOS end-users to install the Vulkan SDK before using Datoviz might be a short-term workaround, but not a good long-term solution. I'm sure there's a good way to do it but I haven't found it yet.